controller/api/destination/endpoint_translator.go

```diff
@@ -310,6 +440,30 @@ func (et *endpointTranslator) sendClientRemove(set watcher.AddressSet) {
 	}
 }
 
+// closeEndpointPolicy closes policy server watches for pod addresses that
+// have been removed. If there is a policy server client for an address in the
+// set, it is closed and then removed from the policyWatches map.
+func (et *endpointTranslator) closeEndpointPolicy(set watcher.AddressSet) {
+	for _, addr := range set.Addresses {
+		// If the address is not backed by a pod then there will be no policy
+		// server watch to close.
+		if addr.Pod == nil {
+			continue
+		}
+		portSpec := portSpec{
+			workload: fmt.Sprintf("%s:%s", addr.Pod.Namespace, addr.Pod.Name),
+			port:     addr.Port,
+		}
+		if client, ok := et.policyWatches[portSpec]; ok {
+			err := client.CloseSend()
```

Comment:
Reading a little more about this function online it doesn't seem like it's the right thing to use here even though I was observing `Recv` errors on the other end; it looks like it still leaves some resources behind. Instead of mapping to policy server clients, I created a `watchPortClient` wrapper that includes the `cancel` function for the context and we use that now.